test/statement_test.ts

```diff
@@ -30,9 +30,9 @@ describe('statements', () => {
   });
   it('translates try/catch', () => {
     expectTranslate('try {} catch(e) {} finally {}')
-        .to.equal(' try { } catch ( e ) { } finally { }');
+        .to.equal(' try { } catch ( e , e_stack ) { } finally { }');
```

Comment:
I would be concerned that it's hard to enforce using the facade since
coding the TS idiom would only break error reporting.
Why not code in idiomatic TypeScript and transpile from your second example
to the first? We just need to understand the types (will have soon)

On Sun, May 10, 2015, 12:44 PM Yegor <notifications@github.com> wrote:

> In test/statement_test.ts
> <https://github.com/angular/ts2dart/pull/153#discussion_r30005268>:
>
> > @@ -30,9 +30,9 @@ describe('statements', () => {
> >    });
> >    it('translates try/catch', () => {
> >      expectTranslate('try {} catch(e) {} finally {}')
> > -        .to.equal(' try { } catch ( e ) { } finally { }');
> > +        .to.equal(' try { } catch ( e , e_stack ) { } finally { }');
>
> To be precise, what I'm proposing long-term is that we use the common
> syntax in both Dart and JS and access stack via a facade (because the field
> names are different; it's err.stack vs err.stackTrace). Example:
>
> // facade/error.dartclass ErrorWrapper {
>   static stackTrace(Error err) => err.stackTrace;
> }
>
> // facade/error.tsexport class ErrorWrapper {
>   static stackTrace(err:Error) {
>     return err.stack;
>   }
> }
>
> Then in all places in Angular 2 code we would:
>
> try {
>   ...
> } catch(err) {
>   var st = ErrorWrapper.stackTrace(err);
> }
>
> Works in both languages.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/ts2dart/pull/153/files#r30005268>.
>